docs: add tutorial on building executor

cristianmtr opened the pull request:
Closes https://github.com/jina-ai/internal-tasks/issues/171

alexcg1 reviewed (approved):
LGTM :+1:

Still needs some rewording to make it fit with other docs, but right now I want to get it out there. I can nitpick later :)

hanxiao commented on docs/fundamentals/tutorials/index.md:
````diff
@@ -0,0 +1,10 @@
+These guide you towards achieving a specific goal in Jina.
+
+```{toctree}
+
````
```suggestion
:hidden:
```

hanxiao commented on docs/index.md:
```diff
@@ -99,7 +99,7 @@ fundamentals/concepts
 fundamentals/executor/index
 fundamentals/flow/index
-fundamentals/practice-your-learning
+fundamentals/tutorials/index
```
tutorials should not be part of fundamentals, as it may contain advanced topics in the future.

Please move it to a new top-level section.

alaeddine-13 replied:
maybe you can merge this branch: docs-chatbot-tutorial to yours and add the executor tutorial to the same section

alaeddine-13 replied:
Ignore the comment above, I'll merge yours  so that both tutorials fall in the same section

hanxiao reviewed (changes requested):
tutorials should not be part of fundamentals, as they may contain advanced topics in the future.

Please move it to a new top-level section.

alaeddine-13 commented on docs/fundamentals/tutorials/new_executor.md:
````diff
@@ -0,0 +1,370 @@
+pip install jina
+ ```
+
+For more information on installing Jina, refer to this [page](https://github.com/jina-ai/jina#install).
````
````suggestion
```{admonition} See Also
:class: seealso
For more information on installing Jina, refer to the {ref}`installation page <installation>`
```
````
and in the docs installation section (get-started/install.md) do this change:
```
(installation)=
# Install
```

alaeddine-13 commented on docs/fundamentals/tutorials/new_executor.md:
````diff
@@ -0,0 +1,370 @@
+
+If you would like to call your `log` function only on `/index` time, you specify the endpoint with `on=`, like this:
+
+```python
````
````suggestion
```{code-block} python
---
emphasize-lines: 1
---
````

alaeddine-13 commented on docs/fundamentals/tutorials/new_executor.md:
````diff
@@ -0,0 +1,370 @@
+:align: center
+```
+
+Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`. 
````
````suggestion
```{admonition} Note
:class: note
Since we pushed our Executor using the `--public` flag, the only thing we will use is the ID, which, in this case, is `zsor7fe6`. 
```
````
Maybe also explain what does public mean and reference the push to hub section

alaeddine-13 reviewed (commented):
I also see many references done like this: `[the docs](https://docs.jina.ai/fundamentals/executor/executor-built-in-features/#handle-request-parameters)`
I personally prefer using this method : https://myst-parser.readthedocs.io/en/latest/sphinx/intro.html#reference-a-section-label-with-a-role
because it doesn't depend on the docs domain name, less prone to be broken and people who modify sections will be aware that it's referenced

cristianmtr commented:
@hanxiao I moved the tutorial to their own top-level section. It now looks like this

![Screenshot from 2021-09-10 10-31-18](https://user-images.githubusercontent.com/8330330/132825020-baa048e0-af8b-4541-b1d7-4f42abe1c954.png)

@alaeddine-13 

I refactored the links into `{ref}`s. Cool feature :) 

Please re-check

alaeddine-13 reviewed (approved):
LGTM

hanxiao commented on docs/tutorials/practice-your-learning.md:
```diff
@@ -1,4 +1,4 @@
-# Put into Practice
+# Putting It All Together
```
please revert this change and make this file stays in the original position

cristianmtr replied:
Done
